models/plan.py

```diff
@@ -0,0 +1,36 @@
+from collections import Iterable
+from datetime import date
+from enum import Enum
+
+from models.camp import Camp
+
+
+class Plan:
+    class EmergencyType(Enum):
+        EARTHQUAKE = 'earthquake'
+        FIRE = 'fire'
+        TSUNAMI = 'tsunami'
+        STORM = 'storm'
+
+    def __init__(self,
+                 name: str,
+                 emergency_type: EmergencyType,
+                 description: str,
+                 geographical_area: str,
+                 start_date: date,
+                 camps: Iterable[Camp] = None):
+        self.name = name
+        self.emergency_type = emergency_type
+        self.description = description
+        self.geographical_area = geographical_area
+        self.start_date = start_date
+        self.camps: set[Camp] = set(camps) if camps else set()
```

Comment:
Should be mandatory to add at least one camp at initialisation

Reply:
If a camp is not included, the following exception can be raised:

```
    def missing_camps_error(Exception):
        """
        It is mandatory to supply at least one camp when you are creating an emergency plan.
        When no camps are provided, this exception will be raised.
        """

        def __init__(self, camps=set(), message="It is mandatory to provide at least one camp"):
            self.camps = camps
            self.message = message
            super().__init__(self.message)
```

Reply:
@Elynor-Kamil I think you mean a class.
I'd suggest removing the camps attribute, and make the message fixed. 
i.e. 
```
    class MissingCampsError(Exception):
        """
        It is mandatory to supply at least one camp when you are creating an emergency plan.
        When no camps are provided, this exception will be raised.
        """

        def __init__(self):
            super().__init__("It is mandatory to provide at least one camp")
```
Is that okay? If so I will add it to the code